config-store-api/src/main/java/io/stargate/config/store/api/ConfigWithOverrides.java

```diff
@@ -36,6 +39,84 @@ public ConfigWithOverrides(@Nonnull Map<String, Object> configMap, String module
     return configMap;
   }
 
+  /**
+   * It retrieves the list of values from underlying configMap checking for its presence and the
+   * type. If the value for a given settingName does not have a value, it throws an {@code
+   * IllegalArgumentException}. If the type of the setting value is not a {@link List} it throws
+   * {@code IllegalArgumentException}. If the underlying type of any element in this list does not
+   * match the expectedType, it throws {@code IllegalArgumentException}.
+   *
+   * @param settingName - it will be used as a key of underlying configMap.
+   * @param expectedType - it will be used to check if the type of an actual list element value
+   *     matches.
+   * @return the list of values associated with settingName. Each value matches expectedType.
+   */
+  @Nonnull
+  @SuppressWarnings("unchecked")
+  public <T> List<T> getSettingValueList(String settingName, Class<T> expectedType) {
+    List<?> settingValue = getSettingValue(settingName, List.class);
+    return settingValue.stream()
+        .map(
+            v -> {
+              validateType(String.format("%s.list-value", settingName), v, expectedType);
+              return (T) v;
+            })
+        .collect(Collectors.toList());
```

Comment:
`Collections.checkedList`?

Reply:
It will look like this:
```
    List settingValue = getSettingValue(settingName, List.class);
	settingValue.forEach(v -> validateType(String.format("%s.list-value", settingName), v, expectedType));
    return Collections.checkedList(settingValue, expectedType);
```
please note that I cannot do `List<?> settingValue = getSettingValue(settingName, List.class);`
because then `checkedList()` will not work.
I am not sure which version looks better. I am ok with both. wdyt?

Reply:
this is a minor point. Feel free to go ahead with current version :+1: